### ISSUE TITLE
Robolectric enters infinite loop in the scheduler

### ISSUE THREAD

**Issue by mikesol** [robolectric/robolectric#3193]
### Description

Robolectric is currently in an infinite loop in the scheduler under certain conditions.

### Steps to Reproduce

My debugger shows that  the following loop in `Scheduler.java` is executing without ever advancing removing the task from the list, leading `this.nextTaskIsScheduledBefore` to return `true` indefinitely.

```
 for(runCount = 0; this.nextTaskIsScheduledBefore(endTime); ++runCount) {
                this.runOneTask();
            }
```

The debugger suggests that it is linked to scrolling - the loop causes an additional element to always be added to the runnable and they alternate between posting a message to the handler and calling `GapWorker`.
### Robolectric & Android Version
3.3 (robolectric) targeting sdk 23

**Comment by mikesol** [comment]
I wound up being able to move past the bug by calling:
```    rv?.isLayoutFrozen = true
```
on my recycler view, which avoids the infinite scrolling loop problem. This could be posted as a workaround for dealing with recycler views if you just want to click on a few items, not unlike the:
```
    rv?.measure(0, 0)
    rv?.layout(0, 0, insanelyBig, insanelyBig)
```
workaround proposed elsewhere to get the recycler view to stretch to a testable size.

**Comment by virl** [comment]
For architecturally correct fix proposal see #2977

**Comment by utzcoz** [comment]
Please try the latest Robolectric, and reopen it if it still exists.
